server/src/main/kotlin/io/provenance/invoice/services/InvoiceService.kt

```diff
@@ -22,11 +23,10 @@ class InvoiceService(
 ) {
     private companion object : KLogging()
 
-    // TODO: Return onboarding API results for the invoice instead of just validate / insert / return
     fun onboardInvoice(request: OnboardInvoiceRequest): OnboardInvoiceResponse {
         // Ensure first that the invoice is of a valid format
         logger.info("Validating received invoice with uuid [${request.invoice.invoiceUuid.value}] has not yet been boarded")
-        InvoiceValidator.validateInvoice(request.invoice)
+        ValidatedInvoice.new(request.invoice).generateValidationReport().throwFailures()
```

Comment:
Implementation of my new Arrow-based validation system for invoices.  It's a bit more verbose, but the gainz from it are awesome